#Tugas Kelompok


Nama Anggota :
1. Achmad Nurs Syururi Arifin
2. Anggie Maesyaroh
3. Devita Hasnasya Rahma
4. Lu'lu' Mudhiah Adilah
5. Muhammad Rizky Irawan Wijaya
6. Retno Sari Sirotujanah


#SOAL NO.1


In [ ]:
import pandas as pd

# menggunakan link github untuk menampilan datasheet
url = 'https://raw.githubusercontent.com/luluma19/kompi2/main/df_2.csv'

df = pd.read_csv(url)

# menampilkan data
print(df.to_string(index=False))


          Timestamp No. Peserta                          Nama                  Jurusan  Berat Badan (Kg)  Tinggi Badan (Cm)
12/09/2024 22:10:42     DS70222        Retno Sari Sirotujanah              Informatika                63                160
12/09/2024 22:12:25     DS70205    Aldy Iriansyah Syarifuddin       Teknik Informatika                55                168
12/09/2024 22:12:26     DS70218  Muhammad Rizki Irawan Wijaya            Ilmu Komputer               100                176
12/09/2024 22:12:30     DS70202    Achmad Nurs Syururi Arifin D4 Manajemen Informatika                68                180
12/09/2024 22:13:33     DS70209               Crisvane Dinda             Ilmu Ekonomi                 70                160
12/09/2024 22:17:49     DS70217             Melati Anggiasari               Statistika                46                156
12/09/2024 22:17:59     DS70206              Anggie Maesyaroh              Informatika                78                162
12/09/20

Kode di atas menggunakan pustaka pandas untuk membaca file CSV dari URL dan mencetaknya tanpa menampilkan index. Pertama, file CSV dibaca dari URL yang berasal dari url github agar tidak perlu mengupload ulang datasheet csv setiap membuka file ini.

menggunakan `pd.read_csv()`, yang mengubahnya menjadi DataFrame. Secara default, saat DataFrame ditampilkan dengan` print(df)`, pandas menambahkan kolom index (nomor baris otomatis). Untuk menghilangkan index tersebut dalam output, metode `to_string(index=False)` digunakan.

# Drop Timestamp & Menghapus kata Kg, cm dan Cm


In [ ]:
df = df.drop('Timestamp', axis=1)
print(df.to_string(index=False))

No. Peserta                          Nama                  Jurusan  Berat Badan (Kg)  Tinggi Badan (Cm)
    DS70222        Retno Sari Sirotujanah              Informatika                63                160
    DS70205    Aldy Iriansyah Syarifuddin       Teknik Informatika                55                168
    DS70218  Muhammad Rizki Irawan Wijaya            Ilmu Komputer               100                176
    DS70202    Achmad Nurs Syururi Arifin D4 Manajemen Informatika                68                180
    DS70209               Crisvane Dinda             Ilmu Ekonomi                 70                160
    DS70217             Melati Anggiasari               Statistika                46                156
    DS70206              Anggie Maesyaroh              Informatika                78                162
    DS70221            Puteri Elisabeth P       Teknik Informatika                54                160
    DS70219                  Nazwa Nabila         Sistem Informa

Kode `df = df.drop('Timestamp', axis=1)` digunakan untuk menghapus kolom bernama 'Timestamp' dari DataFrame `df`. Metode `drop()` mengambil dua argumen utama: nama kolom yang akan dihapus ('Timestamp') dan `axis=1`, yang menunjukkan bahwa yang ingin dihapus adalah kolom (bukan baris, yang akan menggunakan `axis=0`). Setelah kolom dihapus, DataFrame yang diperbarui disimpan kembali ke dalam variabel `df`. Kemudian, `print(df.to_string(index=False))` digunakan untuk menampilkan DataFrame yang telah dimodifikasi tanpa kolom 'Timestamp' dan tanpa index bawaan.

In [ ]:
# Menghapus "kg" dari kolom 'berat'
df['Berat Badan (Kg)'] = df['Berat Badan (Kg)'].astype(str).str.replace(r'\D+', '', regex=True)

# Menghapus "cm" dari kolom 'tinggi'
df['Tinggi Badan (Cm)'] = df['Tinggi Badan (Cm)'].astype(str).str.replace(r'\D+', '', regex=True)

# Mengonversi kolom menjadi tipe numerik setelah menghapus satuan
df['Berat Badan (Kg)'] = pd.to_numeric(df['Berat Badan (Kg)'], errors='coerce')
df['Tinggi Badan (Cm)'] = pd.to_numeric(df['Tinggi Badan (Cm)'], errors='coerce')

print("\nData setelah dibersihkan:")
print(df.to_string(index=False))


Data setelah dibersihkan:
No. Peserta                          Nama                  Jurusan  Berat Badan (Kg)  Tinggi Badan (Cm)
    DS70222        Retno Sari Sirotujanah              Informatika                63                160
    DS70205    Aldy Iriansyah Syarifuddin       Teknik Informatika                55                168
    DS70218  Muhammad Rizki Irawan Wijaya            Ilmu Komputer               100                176
    DS70202    Achmad Nurs Syururi Arifin D4 Manajemen Informatika                68                180
    DS70209               Crisvane Dinda             Ilmu Ekonomi                 70                160
    DS70217             Melati Anggiasari               Statistika                46                156
    DS70206              Anggie Maesyaroh              Informatika                78                162
    DS70221            Puteri Elisabeth P       Teknik Informatika                54                160
    DS70219                  Nazwa Na

Kode di atas melakukan pembersihan dan konversi data pada kolom 'Berat Badan (Kg)' dan 'Tinggi Badan (Cm)' di DataFrame `df`. Pertama, dengan menggunakan `astype(str).str.replace(r'\D+', '', regex=True)`, kode ini menghapus semua karakter non-digit (seperti "kg" dan "cm") dari kedua kolom, sehingga hanya angka yang tersisa dalam bentuk string. Selanjutnya, kedua kolom tersebut diubah menjadi tipe numerik menggunakan `pd.to_numeric()`, di mana `errors='coerce'` digunakan untuk memastikan bahwa jika terdapat nilai yang tidak bisa dikonversi menjadi angka, maka nilai tersebut akan diubah menjadi `NaN` (Not a Number). Akhirnya, hasil dari pembersihan dan konversi ditampilkan dengan mencetak DataFrame yang telah dibersihkan, memungkinkan pengguna untuk melihat data yang sudah siap untuk analisis lebih lanjut.

#Menentukan Rata-rata Tinggi Badan dan Berat Badan


In [ ]:
df.columns = ['No. Peserta', 'Nama', 'Jurusan', 'Berat Badan', 'Tinggi Badan']

rata_mean = df['Tinggi Badan'].mean() #Menghitung rata-rata Tinggi Badan
berat_mean = df['Berat Badan'].mean() #Menghitung rata-rata Berat Badan

#print hasil
print(f"Tinggi Badan Rata-Rata : {rata_mean:.2f} cm ") #Menggunakan f-string untuk mencetak teks dengan nilai rata_mean, Format :.2f artinya angka rata-rata tinggi badan akan ditampilkan dengan dua angka di belakang koma, dan satuan yang ditampilkan adalah cm.
print(f"Berat Badan Rata-Rata : {berat_mean:.2f} kg") #Ditampilkan dalam satuan Kg

Tinggi Badan Rata-Rata : 162.79 cm 
Berat Badan Rata-Rata : 61.12 kg


1. Codingan diatas digunakan untuk menghitung rata-rata berat dan dan tinggi badan lalu menampilkan hasilnya dengan format yang sudah ditentukan
2. output yang dihasilkan menampilkan rata-rata tersebut denganformat dua angka desm=imal menggunakan f-string

#Identifikasi Type Data dan Jumlah baris&kolom

In [ ]:
#Menampilkan Tipe Data
print(df.dtypes)

No. Peserta          object
Nama                 object
Jurusan              object
Berat Badan (Kg)      int64
Tinggi Badan (Cm)     int64
dtype: object


Kode print(df.dtypes) adalah atribut dalam objek DataFrame di pustaka pandas yang digunakan untuk menampilkan tipe data dari setiap kolom dalam DataFrame tersebut

In [ ]:
#Menampilkan Baris dan Kolom
print(df.shape)

(25, 5)


de print(df.shape) digunakan untuk menampilkan jumlah baris dan kolom yang terdapat dalam DataFrame df. Atribut shape mengembalikan sebuah tuple yang terdiri dari dua nilai: nilai pertama adalah jumlah baris, dan nilai kedua adalah jumlah kolom.

In [ ]:
#Menampilkan informasi dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   No. Peserta        25 non-null     object
 1   Nama               25 non-null     object
 2   Jurusan            25 non-null     object
 3   Berat Badan (Kg)   25 non-null     int64 
 4   Tinggi Badan (Cm)  25 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1.1+ KB


#2. Hitung BMI dalam dataset


Hitunglah BMI setiap siswa dengan rumus BMI = **Berat Badan(Kg)/(Tinggi Badan ((cm))²** , masukkan hasil perhitungan ke dalam dataframe.

In [ ]:
#Fungsi Hitung BMI
def Hitung_BMI(df) :
  #Konversi Tinggi Badan ke dalam Meter
  df['Tinggi Badan (M)'] = df['Tinggi Badan (Cm)']/100

  #Program BMI
  df['BMI'] = df['Berat Badan (Kg)'] / (df['Tinggi Badan (M)']**2)

  #Menampilkan BMI dengan 2 angka di belakang koma
  df['BMI'] = df['BMI'].round(2)
  return df

#Tambah Hitung BMI ke Dataset
df = Hitung_BMI(df)

print(df)


   No. Peserta                           Nama                   Jurusan  \
0      DS70222         Retno Sari Sirotujanah               Informatika   
1      DS70205     Aldy Iriansyah Syarifuddin        Teknik Informatika   
2      DS70218   Muhammad Rizki Irawan Wijaya             Ilmu Komputer   
3      DS70202     Achmad Nurs Syururi Arifin  D4 Manajemen Informatika   
4      DS70209                Crisvane Dinda              Ilmu Ekonomi    
5      DS70217              Melati Anggiasari                Statistika   
6      DS70206               Anggie Maesyaroh               Informatika   
7      DS70221             Puteri Elisabeth P        Teknik Informatika   
8      DS70219                   Nazwa Nabila          Sistem Informasi   
9      DS70212                         Amalia             Ilmu Komputer   
10     DS70224                  Stepanus Malo        Teknik Informatika   
11     DS70220   Pandu Giswantoro Abhi Wiguno       Teknologi Informasi   
12     DS70213           

1. Kodingan tersebut digunakan untuk menghitung BMI (Body Mass Index) dari data berat badan dan tinggi badan lalu menampilkan hasilnya
2. BMI dihitung berdasarkan rumus berat badan yang dibagi dengan tinggi badan dalam meter dikuadratkan
3. nilai BMI dibulatkan hingga 2 angka dibelakang koma
4. Nalai akhir menampilkan nama peserta dan nama BMI mereka dalam bentuk tabel nama menampilkan nomor baris


#3. Membuat Program untuk menghitung BMI



a. Buatlah function dengan input berat badan dan tinggi badan dari user (dalam decimal)

b. Secara otomatis program akan menghitung BMI.

In [ ]:
1# Fungsi untuk menghitung BMI
def BMI_Calculator():
    try:
        # Meminta input dari user untuk tinggi badan (dalam cm)
        print("Masukkan Tinggi Badan (cm):")
        Tinggi_Badan = float(input())  # Mengonversi input ke tipe float

        # Meminta input dari user untuk berat badan (dalam kg)
        print("Masukkan Berat Badan (kg):")
        Berat_Badan = float(input())  # Mengonversi input ke tipe float

        # Konversi tinggi badan dari cm ke meter
        Tinggi_Badan_M = Tinggi_Badan / 100

        # Menghitung BMI
        BMI = Berat_Badan / (Tinggi_Badan_M ** 2)

        # Menampilkan hasil BMI
        print(f"Nilai BMI Anda adalah: {BMI:.2f}")

    except ValueError:
        print("Input tidak valid. Pastikan Anda memasukkan angka.")


# Memanggil fungsi BMI_Calculator untuk menjalankan program
BMI_Calculator()


Masukkan Tinggi Badan (cm):
160
Masukkan Berat Badan (kg):
60
Nilai BMI Anda adalah: 23.44


1. Codingan ini menerima input berat badan dan tinggi badan dari pengguna
2. BMI dihitung menggunakan fungsi (hitung_bmi)
3. Hasil BMI ditampilkan dengan 2 angka desimal


#4. Menampilkan status dalam dataset


Dari dataframe soal No. 2, buatlah status BMI untuk setiap siswa dari kategori berikut, lalu simpan data statusnya dengan nama kolom Status BMI.

Normal: 18.5-24.9

Overweight: 25.0-29.9

Obesity: 30.0-34.9

Underweight: <18.5

#Menggunakan IF ELSE

In [ ]:
#Fungsi Status_BMI
def Status_BMI(df):
    Status_BMI = []
    for BMI in df['BMI']:
        if BMI < 18.5:
            Status_BMI.append('Underweight')
        elif 18.5 <= BMI <= 24.9:
            Status_BMI.append('Normal')
        elif 25.0 <= BMI <= 29.9:
            Status_BMI.append('Overweight')
        elif BMI >= 30.0:
            Status_BMI.append('Obesity')
        else:
            Status_BMI.append('Unknown')

    df['Status BMI'] = Status_BMI
    return df

# Menambahkan Status BMI ke DataFrame
df = Status_BMI(df)

# Menampilkan DataFrame
print(df)

   No. Peserta                           Nama                   Jurusan  \
0      DS70222         Retno Sari Sirotujanah               Informatika   
1      DS70205     Aldy Iriansyah Syarifuddin        Teknik Informatika   
2      DS70218   Muhammad Rizki Irawan Wijaya             Ilmu Komputer   
3      DS70202     Achmad Nurs Syururi Arifin  D4 Manajemen Informatika   
4      DS70209                Crisvane Dinda              Ilmu Ekonomi    
5      DS70217              Melati Anggiasari                Statistika   
6      DS70206               Anggie Maesyaroh               Informatika   
7      DS70221             Puteri Elisabeth P        Teknik Informatika   
8      DS70219                   Nazwa Nabila          Sistem Informasi   
9      DS70212                         Amalia             Ilmu Komputer   
10     DS70224                  Stepanus Malo        Teknik Informatika   
11     DS70220   Pandu Giswantoro Abhi Wiguno       Teknologi Informasi   
12     DS70213           

1. Fungsi status_BMI menghitung status bmi untuk setiap indivisu berdasarkan nilai BMI mereka.
2. Program ini menampilkan dataframe lengkap dengan status BMI setelah perhitungan

#Menggunakan np.where

In [ ]:
import numpy as np

#Fungsi Status_BMI
def Status_BMI(df):
    df['Status_BMI'] =    np.where(df['BMI'] < 18.5, 'Underweight',
                          np.where((df['BMI'] >= 18.5) & (df['BMI'] <= 24.9), 'Normal',
                          np.where((df['BMI'] >= 25.0) & (df['BMI'] <= 29.9), 'Overweight',
                          np.where(df['BMI'] >= 30.0, 'Obesity', 'Unknown'))))
    return df

# Menambahkan Status BMI ke DataFrame
df = Status_BMI(df)

# Menampilkan DataFrame
print(df)

   No. Peserta                           Nama                   Jurusan  \
0      DS70222         Retno Sari Sirotujanah               Informatika   
1      DS70205     Aldy Iriansyah Syarifuddin        Teknik Informatika   
2      DS70218   Muhammad Rizki Irawan Wijaya             Ilmu Komputer   
3      DS70202     Achmad Nurs Syururi Arifin  D4 Manajemen Informatika   
4      DS70209                Crisvane Dinda              Ilmu Ekonomi    
5      DS70217              Melati Anggiasari                Statistika   
6      DS70206               Anggie Maesyaroh               Informatika   
7      DS70221             Puteri Elisabeth P        Teknik Informatika   
8      DS70219                   Nazwa Nabila          Sistem Informasi   
9      DS70212                         Amalia             Ilmu Komputer   
10     DS70224                  Stepanus Malo        Teknik Informatika   
11     DS70220   Pandu Giswantoro Abhi Wiguno       Teknologi Informasi   
12     DS70213           

#5. Mengurutkan Tinggi badan dari tertinggi ke terendah


a. Salin dataframe no. 4 dan beri nama df_tinggi, lalu urutkan data berdasarkan tinggi badan dari yang tertinggi ke rendah.



In [ ]:
df_tinggi = df[['Nama', 'Jurusan','Tinggi Badan (Cm)']].copy()

df_tinggi = df_tinggi.sort_values(by='Tinggi Badan (Cm)', ascending=False)

# Menampilkan DataFrame df_tinggi
print(df_tinggi)

                             Nama                   Jurusan  Tinggi Badan (Cm)
3      Achmad Nurs Syururi Arifin  D4 Manajemen Informatika                180
16                   Fiari Rahman        Teknik Informatika                178
2    Muhammad Rizki Irawan Wijaya             Ilmu Komputer                176
13                    Abdul Hafiz               Informatika                175
11   Pandu Giswantoro Abhi Wiguno       Teknologi Informasi                172
15            Afif Fadhil Mahmudi        Teknik Informatika                171
1      Aldy Iriansyah Syarifuddin        Teknik Informatika                168
14                Ayrisa Trianida               Informatika                165
17           Cisa Nada Salsabilla     Pendidikan Matematika                163
12                           Lisa              Ilmu Ekonomi                162
22  M. Riziq Sirfatullah Alfarizi             ilmu komputer                162
6                Anggie Maesyaroh               Info

1. kodingan ini akan mengsortir kolom tinggi badan
2. hasil yang diatampilkan data tinggi badan dari tertinggi hingga terendah

#Mengurutkan berdasarkan nama dari Z-A

In [ ]:
df_tinggi = df[['Nama', 'Jurusan']].copy()

df_tinggi = df_tinggi.sort_values(by='Nama', ascending=False)

# Menampilkan DataFrame df_tinggi
print(df_tinggi)

                             Nama                   Jurusan
20                 Taufiqurrohman          Sistem Informasi
10                  Stepanus Malo        Teknik Informatika
24                 Safinatunnajah        Teknik Informatika
0          Retno Sari Sirotujanah               Informatika
7              Puteri Elisabeth P        Teknik Informatika
11   Pandu Giswantoro Abhi Wiguno       Teknologi Informasi
8                    Nazwa Nabila          Sistem Informasi
2    Muhammad Rizki Irawan Wijaya             Ilmu Komputer
5               Melati Anggiasari                Statistika
18                 Meilisa Triana          Sistem Informasi
22  M. Riziq Sirfatullah Alfarizi             ilmu komputer
19         Lu'lu' Mudhiah Adilah        Teknik Informatika 
12                           Lisa              Ilmu Ekonomi
16                   Fiari Rahman        Teknik Informatika
21          Devita Hasnasya Rahma          Sistem Informasi
4                 Crisvane Dinda        

1. kodingan ini akan mengsortir kolom nama
2. hasil yang ditampilakan nama dengan urutan abjad dari Z-A

# 6. Filter status dataframe

# a. Status BMI diluar Normal

In [ ]:
import pandas as pd

#Fungsi Status_BMI
def Status_BMI(df):
    Status_BMI = []
    for BMI in df['BMI']:
        if BMI < 18.5:
            Status_BMI.append('Underweight')
        elif 18.5 <= BMI <= 24.9:
            Status_BMI.append('Normal')
        elif 25.0 <= BMI <= 29.9:
            Status_BMI.append('Overweight')
        elif BMI >= 30.0:
            Status_BMI.append('Obesity')
        else:
            Status_BMI.append('Unknown')

    df['Status BMI'] = Status_BMI
    return df

# Menambahkan Status BMI ke DataFrame
df = Status_BMI(df)

# Filter untuk mendapatkan orang dengan BMI lebih dari Normal
df_Overweight = df[df['BMI'] > 24.9]

# Menampilkan DataFrame yang difilter
print("Orang dengan BMI lebih dari 25:")
print(df_Overweight)


Orang dengan BMI lebih dari 25:
   No. Peserta                          Nama              Jurusan  \
2      DS70218  Muhammad Rizki Irawan Wijaya        Ilmu Komputer   
4      DS70209               Crisvane Dinda         Ilmu Ekonomi    
6      DS70206              Anggie Maesyaroh          Informatika   
13     DS70201                   Abdul Hafiz          Informatika   
16     DS70211                  Fiari Rahman   Teknik Informatika   
19     DS70214        Lu'lu' Mudhiah Adilah   Teknik Informatika    
20     DS70225                Taufiqurrohman     Sistem Informasi   

    Berat Badan (Kg)  Tinggi Badan (Cm)  Tinggi Badan (M)    BMI  Status BMI  \
2                100                176              1.76  32.28     Obesity   
4                 70                160              1.60  27.34  Overweight   
6                 78                162              1.62  29.72  Overweight   
13               110                175              1.75  35.92     Obesity   
16             

1. Kodingan ini berfungsi untuk mengfilter dan menampilakan data individu berdasarkan status BMI
2. Hasil yang ditampilan untuk analisis tentang status kesehatan berdasarkan BMI dengan

Underweight: BMI < 18.5

Normal: 18.5 ≤ BMI ≤ 24.9

Overweight: 25.0 ≤ BMI ≤ 29.9

Obesity: BMI ≥ 30.0

#b. Tinggi Badan > 169 cm dan Berat Badan > 60 kg

In [ ]:
import pandas as pd

#Fungsi Status_BMI
def Status_BMI(df):
    Status_BMI = []
    for BMI in df['BMI']:
        if BMI < 18.5:
            Status_BMI.append('Underweight')
        elif 18.5 <= BMI <= 24.9:
            Status_BMI.append('Normal')
        elif 25.0 <= BMI <= 29.9:
            Status_BMI.append('Overweight')
        elif BMI >= 30.0:
            Status_BMI.append('Obesity')
        else:
            Status_BMI.append('Unknown')

    df['Status BMI'] = Status_BMI
    return df

# Menambahkan Status BMI ke DataFrame
df = Status_BMI(df)

# Filter untuk mendapatkan orang dengan Tinggi Badan > 169 Cm & Berat Badan > 60 Kg
df_star = df[(df['Tinggi Badan (Cm)'] > 169) & (df['Berat Badan (Kg)'] > 60)]

# Menampilkan DataFrame yang difilter
print("Orang dengan Tinggi Badan > 169 CM dan Berat Badan > 60 KG:")
print(df_star)

Orang dengan Tinggi Badan > 169 CM dan Berat Badan > 60 KG:
   No. Peserta                          Nama                   Jurusan  \
2      DS70218  Muhammad Rizki Irawan Wijaya             Ilmu Komputer   
3      DS70202    Achmad Nurs Syururi Arifin  D4 Manajemen Informatika   
11     DS70220  Pandu Giswantoro Abhi Wiguno       Teknologi Informasi   
13     DS70201                   Abdul Hafiz               Informatika   
16     DS70211                  Fiari Rahman        Teknik Informatika   

    Berat Badan (Kg)  Tinggi Badan (Cm)  Tinggi Badan (M)    BMI  Status BMI  \
2                100                176              1.76  32.28     Obesity   
3                 68                180              1.80  20.99      Normal   
11                70                172              1.72  23.66      Normal   
13               110                175              1.75  35.92     Obesity   
16                80                178              1.78  25.25  Overweight   

    Status_BMI

1. Kodingan ini memfilter data berdasarkan tinggi badan > 169 dan berat bdan >60 kg
2. Hasil yang ditampilkan nama data yang sudah dimfilter

#c. Nama yang mengandung huruf "y".

In [ ]:
import pandas as pd

#Fungsi Status_BMI
def Status_BMI(df):
    Status_BMI = []
    for BMI in df['BMI']:
        if BMI < 18.5:
            Status_BMI.append('Underweight')
        elif 18.5 <= BMI <= 24.9:
            Status_BMI.append('Normal')
        elif 25.0 <= BMI <= 29.9:
            Status_BMI.append('Overweight')
        elif BMI >= 30.0:
            Status_BMI.append('Obesity')
        else:
            Status_BMI.append('Unknown')

    df['Status BMI'] = Status_BMI
    return df

# Menambahkan Status BMI ke DataFrame
df = Status_BMI(df)

# Filter untuk mendapatkan nama orang yang mengandung huruf "y"
df_huruf = df[df['Nama'].str.contains('y', case=False)]

# Menampilkan DataFrame yang difilter
print("Orang yang namanya mengandung huruf 'y':")
print(df_huruf)


Orang yang namanya mengandung huruf 'y':
   No. Peserta                          Nama                   Jurusan  \
1      DS70205    Aldy Iriansyah Syarifuddin        Teknik Informatika   
2      DS70218  Muhammad Rizki Irawan Wijaya             Ilmu Komputer   
3      DS70202    Achmad Nurs Syururi Arifin  D4 Manajemen Informatika   
6      DS70206              Anggie Maesyaroh               Informatika   
14     DS70207               Ayrisa Trianida               Informatika   
21     DS70210         Devita Hasnasya Rahma          Sistem Informasi   

    Berat Badan (Kg)  Tinggi Badan (Cm)  Tinggi Badan (M)    BMI  Status BMI  \
1                 55                168              1.68  19.49      Normal   
2                100                176              1.76  32.28     Obesity   
3                 68                180              1.80  20.99      Normal   
6                 78                162              1.62  29.72  Overweight   
14                58                165 

1. Kodingan memfilter berdasarkan nama yang mengandung hruf 'y'
2. Hasil kodingan menampilakan data yang mengandung huruf 'y'

#d. Siapa saja yang berat badannya kurang dari 60 kg?

In [ ]:
import pandas as pd

#Fungsi Status_BMI
def Status_BMI(df):
    Status_BMI = []
    for BMI in df['BMI']:
        if BMI < 18.5:
            Status_BMI.append('Underweight')
        elif 18.5 <= BMI <= 24.9:
            Status_BMI.append('Normal')
        elif 25.0 <= BMI <= 29.9:
            Status_BMI.append('Overweight')
        elif BMI >= 30.0:
            Status_BMI.append('Obesity')
        else:
            Status_BMI.append('Unknown')

    df['Status BMI'] = Status_BMI
    return df

# Menambahkan Status BMI ke DataFrame
df = Status_BMI(df)

# Filter untuk mendapatkan orang dengan Berat Badan < 60 Kg
df_weight = df[(df['Berat Badan (Kg)'] < 60)]

# Menampilkan DataFrame yang difilter
print("Orang dengan Tinggi Badan Berat Badan < 60 KG:")
print(df_weight)

Orang dengan Tinggi Badan Berat Badan < 60 KG:
   No. Peserta                           Nama             Jurusan  \
1      DS70205     Aldy Iriansyah Syarifuddin  Teknik Informatika   
5      DS70217              Melati Anggiasari          Statistika   
7      DS70221             Puteri Elisabeth P  Teknik Informatika   
8      DS70219                   Nazwa Nabila    Sistem Informasi   
9      DS70212                         Amalia       Ilmu Komputer   
10     DS70224                  Stepanus Malo  Teknik Informatika   
12     DS70213                           Lisa        Ilmu Ekonomi   
14     DS70207                Ayrisa Trianida         Informatika   
15     DS70203            Afif Fadhil Mahmudi  Teknik Informatika   
18     DS70216                 Meilisa Triana    Sistem Informasi   
21     DS70210          Devita Hasnasya Rahma    Sistem Informasi   
22     DS70215  M. Riziq Sirfatullah Alfarizi       ilmu komputer   
23     DS70204             Afsha Alifia Putri  Teknik In

1. Kodingan memfilter data yang memiliki berat badan < 60
2. output yang dihasilkan nama-nama yang memiliki berat badan < 60

#7. Menambahkan nomor urut

In [ ]:
import pandas as pd

#Fungsi Status_BMI
def Status_BMI(df):
    Status_BMI = []
    for BMI in df['BMI']:
        if BMI < 18.5:
            Status_BMI.append('Underweight')
        elif 18.5 <= BMI <= 24.9:
            Status_BMI.append('Normal')
        elif 25.0 <= BMI <= 29.9:
            Status_BMI.append('Overweight')
        elif BMI >= 30.0:
            Status_BMI.append('Obesity')
        else:
            Status_BMI.append('Unknown')

    df['Status BMI'] = Status_BMI
    return df

# Menambahkan Status BMI ke DataFrame
df = Status_BMI(df)

# Menambahkan kolom 'No.' di bagian paling kiri
df.insert(0, 'No.', range(1, len(df) + 1))

# Menampilkan DataFrame
print(df)

    No. No. Peserta                           Nama                   Jurusan  \
0     1     DS70222         Retno Sari Sirotujanah               Informatika   
1     2     DS70205     Aldy Iriansyah Syarifuddin        Teknik Informatika   
2     3     DS70218   Muhammad Rizki Irawan Wijaya             Ilmu Komputer   
3     4     DS70202     Achmad Nurs Syururi Arifin  D4 Manajemen Informatika   
4     5     DS70209                Crisvane Dinda              Ilmu Ekonomi    
5     6     DS70217              Melati Anggiasari                Statistika   
6     7     DS70206               Anggie Maesyaroh               Informatika   
7     8     DS70221             Puteri Elisabeth P        Teknik Informatika   
8     9     DS70219                   Nazwa Nabila          Sistem Informasi   
9    10     DS70212                         Amalia             Ilmu Komputer   
10   11     DS70224                  Stepanus Malo        Teknik Informatika   
11   12     DS70220   Pandu Giswantoro A

1. Kodingan digunakan untuk menambahkan nomot urut
2. data yang ditampilkan adalah data yang ada nomor urut

#8. Mencari nilai-nilai tertentu

#A. Berapa nilai rata-rata tinggi badan siswa


In [ ]:
# Menghitung rata-rata tinggi badan
rata_rata_tinggi = df['Tinggi Badan (Cm)'].mean()

# Menampilkan hasil
print(f"Rata-rata tinggi badan: {rata_rata_tinggi:.2f} cm")

Rata-rata tinggi badan: 162.40 cm


1. Kodingan digunakan untuk mencari rata-rata tinggi bada
2. output yang dihasilkan tinggi badan rata-rata uaitu = 162.40 cm

#B. Berapa nilal rata-rata tinggi badan siswa?

In [ ]:
# Menghitung rata-rata berat badan
rata_rata_berat = df['Berat Badan (Kg)'].mean()

# Menampilkan hasil
print(f"Rata-rata berat badan: {rata_rata_berat:.2f} kg")

Rata-rata berat badan: 60.44 kg


1. Kodingan digunakan untuk mencari rata-rata Berat badan
2. output yang dihasilkan Berat badan rata-rata uaitu = 60.44 Kg

# C. Ada berapa orang jumlah siswa yang tercatat pada dataframe df?

#Menggunakan count

In [ ]:
# Menghitung Jumlah Siswa
Jumlah_Siswa = df['No.'].count()

# Menampilkan hasil
print(f"Jumlah Siswa: {Jumlah_Siswa} Orang")


Jumlah Siswa: 25 Orang


1. kodingan ini digunakan untuk menghitung jumlah siswa yang ada pada data
2. output yang dihasilkan jumlah siswa yaitu = 25 orang

#Manggunakan len

In [ ]:
# Menghitung Jumlah Siswa menggunakan len()
Jumlah_Siswa = len(df)

# Menampilkan hasil
print(f"Jumlah Siswa: {Jumlah_Siswa} Orang")


Jumlah Siswa: 25 Orang


#D. 5 orang dengan nilai BMI tertinggi.

In [ ]:
# Menghitung 5 Siswa dengan BMI Tertinggi
Top_BMI = df.nlargest(5, 'BMI')  # Menggunakan tanda kurung ()

# Menampilkan hasil
print("5 Siswa dengan BMI Tertinggi adalah:")
print(Top_BMI[['Nama', 'BMI']])


5 Siswa dengan BMI Tertinggi adalah:
                            Nama    BMI
13                   Abdul Hafiz  35.92
2   Muhammad Rizki Irawan Wijaya  32.28
6               Anggie Maesyaroh  29.72
4                Crisvane Dinda   27.34
20                Taufiqurrohman  26.37


1. Kodingan digunakan untuk menghitung 5 siswa dengan BMI tertinggi
2. Output yang dihasilkan yaitu 5 orang siswa dengan BMI tertinggi


#E. Ada berapa orang yang tergolong status Underweight, Normal, Overweight, dan Obesity?

In [ ]:
# Menghitung jumlah orang pada setiap status BMI
jumlah_per_status = df['Status BMI'].value_counts()

# Menampilkan hasil
print("Jumlah orang pada setiap status BMI:")
print(jumlah_per_status)

Jumlah orang pada setiap status BMI:
Status BMI
Normal         15
Overweight      5
Underweight     3
Obesity         2
Name: count, dtype: int64


1. Kodingan ini diggunakan untuk menghitung jumlah orang pada setiap status BMI
2. Ouput yang dihasilkan yaitu jumlah orang yang ada pada setiap status BMI

#SOAL 2


Perusahaan investasi jamin-sugih Co kami memberikan penawaran investasi terbaik dengan pembagian keuangan tetap sebesar 25% setiap tahun

Buatkanlah program python dapat menerima input dari pengguna dalam bentuk : Modal investasi (bilangan bulat antara 0 - 1.000.000.000)

Jangka waktu investasi (bilangan bulat antara 1 - 25 dalam tahun)

Program akan menampilkan informasi akumulasi setiap tahun, menggunakan iterasi untuk setiap tahun

In [ ]:
# Soal No 2
# Buatkan Investment Tenue Years setiap tahunnya
# Dengan Modal 0-1.000.000.000 dan Jangka Waktu 1-25 Tahun
def investment_simulation():
    try:
        # Input modal investasi
        print("Masukkan Modal Investasi 0-1.000.000.000")
        capital = int(input("Money Capital : "))
        if capital < 0 or capital > 1_000_000_000:
            print ("Modal tidak sesuai ketentuan.")
            return

        # Input jangka waktu investasi
        print("Masukkan Jangka Waktu Investasi 1-25 Tahun")
        tenure = int(input("Investment Tenure : "))
        if tenure < 1 or tenure > 25:
            print("Tenure tidak sesuai ketentuan.")
            return


        rate_of_return = 0.25
        for year in range(1, tenure + 1):
            capital += capital * rate_of_return
            print(f"Year {year}, your total investment: {capital:.1f}")

    except ValueError as e:
        print(f"Error: {e}")

# Menjalankan simulasi investasi
investment_simulation()

Masukkan Modal Investasi 0-1.000.000.000
Money Capital : 100000000
Masukkan Jangka Waktu Investasi 1-25 Tahun
Investment Tenure : 16
Year 1, your total investment: 125000000.0
Year 2, your total investment: 156250000.0
Year 3, your total investment: 195312500.0
Year 4, your total investment: 244140625.0
Year 5, your total investment: 305175781.2
Year 6, your total investment: 381469726.6
Year 7, your total investment: 476837158.2
Year 8, your total investment: 596046447.8
Year 9, your total investment: 745058059.7
Year 10, your total investment: 931322574.6
Year 11, your total investment: 1164153218.3
Year 12, your total investment: 1455191522.8
Year 13, your total investment: 1818989403.5
Year 14, your total investment: 2273736754.4
Year 15, your total investment: 2842170943.0
Year 16, your total investment: 3552713678.8


1. Kode ini memungkikan pengguna untuk mensimulasikan pertumbuhan investasi mereka berdasarkan modal awal dan jangka waktu investasi yang diinginkan.
2. Output yang dihasilkan pengguna dapat melihat bagaimana modal mereka berkembang dari tahun ke tahun

#SOAL No. 3


In [ ]:
#num  berfungsi untuk memeriksa apakah bilangan ini prima atau bukan
def is_prima(num):
#bilangan prima merupakan bilangan yang yang bersifat asli yang lebih dari 1 dan hanya bisa dibagi oleh 1 dan oleh bilanganya itu sendiri
    if num < 2: #ketika angka yang dimasukkan kurang dari dua maka fungsi akan false
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True #jika angka yang dimasukkan sudah benar maka bilangan terebut merupakan bilangan prima
#generate_prima bertugas untuk menghasilkan bilangan (n)prima pertama
def generate_prima(n):
    prima = []
    num = 2
    while len(prima) < n:
        if is_prima(num): #bilangan prima di cek oleh fungsi is_prime
            prima.append(num)
        num += 1
    return prima

# Input yang dimasukkan oleh pengguna
n = int(input("Masukkan Nilai prima yang anda inginkan : "))

# Memanggil fungsi untuk menghasilkan bilangan prima
prima = generate_prima(n)
# Output yang dihasilkan
print(f"Berikut merupakan {n} bilangan prima pertama yang dihasilkan : {prima}")


Masukkan Nilai prima yang anda inginkan : 15
Berikut merupakan 15 bilangan prima pertama yang dihasilkan : [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]


1. Kodingan digunakan untuk menghitung bilangan prima, dan pengguna dapat memaukkan input yang diinginkan
2. Fungsi yang digunakan is_prima untuk memeriksa keprimaan bilangan dan generate_prima untuk menghasilkan bilangan prima
3. Hasil yang ditampilkan merupakan bilangan pertama yang diinput pengguna

#Soal no.4


#Pola 1

In [ ]:
def pola_1(n):
    for i in range(1, n+1):
        # Print leading spaces
        print(' ' * (n-i), end='')
        # Print stars
        print('* ' * i)

n = int(input("Masukkan nilai n: "))
pola_1(n)

Masukkan nilai n: 10
         * 
        * * 
       * * * 
      * * * * 
     * * * * * 
    * * * * * * 
   * * * * * * * 
  * * * * * * * * 
 * * * * * * * * * 
* * * * * * * * * * 


Penjelasan alur program:
1. Pengguna memasukkan nilai (`n`) sebagai input tinggi.
2. Program menggunakan loop untuk mencetak setiap baris dari segitiga
3. Pada setiap iterasi, program `#Print stars` mencetak spasi untuk memposisikan bintang dengan benar, kemudian mencetak bintang dengan spasi diantaranya.
4. Program mencetak baris-baris dengan jumlah bintang yang meningkat pada setiap baris, dimulai dari 1 hingga `n`.



#Pola 2


In [ ]:
def pola_2(n):
    max_width = 2 * n - 1  # Lebar maksimum pola, berdasarkan jumlah huruf terbanyak di baris terakhir

    # Bagian atas
    for i in range(1, n+1):
        # Tentukan jumlah huruf pada setiap baris dengan pola bilangan ganjil
        jumlah_huruf = 2 * i - 1
        # Huruf yang diulang dengan spasi
        baris_huruf = (chr(64 + i) + ' ') * jumlah_huruf
        # Print baris dengan center
        print(baris_huruf.center(max_width * 2))  # Centering berdasarkan lebar maksimum pola

    # Bagian bawah
    for i in range(n-1, 0, -1):
        # Tentukan jumlah huruf pada setiap baris dengan pola bilangan ganjil
        jumlah_huruf = 2 * i - 1
        # Huruf yang diulang dengan spasi
        baris_huruf = (chr(64 + i) + ' ') * jumlah_huruf
        # Print baris dengan center
        print(baris_huruf.center(max_width * 2))  # Centering berdasarkan lebar maksimum pola

n = int(input("Masukkan nilai n: "))
pola_2(n)

Masukkan nilai n: 20
                                      A                                       
                                    B B B                                     
                                  C C C C C                                   
                                D D D D D D D                                 
                              E E E E E E E E E                               
                            F F F F F F F F F F F                             
                          G G G G G G G G G G G G G                           
                        H H H H H H H H H H H H H H H                         
                      I I I I I I I I I I I I I I I I I                       
                    J J J J J J J J J J J J J J J J J J J                     
                  K K K K K K K K K K K K K K K K K K K K K                   
                L L L L L L L L L L L L L L L L L L L L L L L                 
              M M M M M M M M M

Penjelasan alur program:
1. Pengguna memasukkan nilai `n` sebagai input tinggi.
2. Program menggunakan perhitungan `max_width`, yang dihitung berdasarkan jumlah karakter maksimal yang akan dicetak pada baris terpanjang. Rumusnya adalah `2 * n - 1`, di mana `n` adalah jumlah baris.
3. Program menggunakan dua loop: satu untuk mencetak bagian atas dan satu untuk bagian bawah.
4. Loop pertama (Bagian Atas) mencetak huruf dari 'A' hingga karakter sesuai dengan baris saat ini. Pada setiap iterasi, program akan menghitung jumlah huruf berdasarkan rumus 2 * i - 1 (baris pertama memiliki 1 huruf, baris kedua memiliki 3 huruf, dan seterusnya sama seperti pola bilangan ganjil).
5. Loop kedua mencetak huruf untuk bagian bawah, dengan jumlah huruf yang menurun setelah mencapai puncak (jumlah huruf di baris-baris berikutnya berkurang dari 9 kembali ke 1).
6. Fungsi `chr(64 + i)` digunakan untuk mengkonversi angka ke karakter huruf(misalnya, 65 untuk 'A', 66 untuk 'B', dan seterusnya).
7. Setiap baris yang dicetak di-"center"-kan menggunakan metode `.center()`. Untuk memastikan spasi yang cukup, lebar maksimal dikalikan dengan 2, sehingga ada ruang lebih untuk meratakan pola ditengah layar.




#Pola 3

In [ ]:
def pola_huruf(n):
    max_width = 2 * n - 1  # Lebar maksimum pola, berdasarkan jumlah huruf terbanyak di baris terakhir

    # Bagian atas
    for i in range(1, n+1):
        # Tentukan jumlah huruf pada setiap baris dengan pola bilangan ganjil
        jumlah_huruf = 2 * i - 1
        # Gunakan karakter sesuai urutan ASCII
        karakter = chr(64 + i) if i <= 26 else chr(64 + i - 26)
        # Huruf yang diulang dengan spasi
        baris_huruf = (karakter + ' ') * jumlah_huruf
        # Print baris dengan center
        print(baris_huruf.center(max_width * 2))

    # Bagian bawah
    for i in range(n-1, 0, -1):
        # Tentukan jumlah huruf pada setiap baris dengan pola bilangan ganjil
        jumlah_huruf = 2 * i - 1
        # Gunakan karakter sesuai urutan ASCII
        karakter = chr(64 + i) if i <= 26 else chr(64 + i - 26)
        # Huruf yang diulang dengan spasi
        baris_huruf = (karakter + ' ') * jumlah_huruf
        # Print baris dengan center
        print(baris_huruf.center(max_width * 2))

n = int(input("Masukkan nilai n: "))
pola_huruf(n)

Masukkan nilai n: 25
                                                A                                                 
                                              B B B                                               
                                            C C C C C                                             
                                          D D D D D D D                                           
                                        E E E E E E E E E                                         
                                      F F F F F F F F F F F                                       
                                    G G G G G G G G G G G G G                                     
                                  H H H H H H H H H H H H H H H                                   
                                I I I I I I I I I I I I I I I I I                                 
                              J J J J J J J J J J J J J J J J J J J                     

Penjelasan Alur Program:
1. Pengguna memasukkan nilai `n` sebagai input tinggi.
2. Program menggunakan perhitungan `max_width` untuk menentukan lebar maksimal dari pola, yang dihitung dari lebar baris terpanjang.
3. Program menggunakan dua loop: satu untuk mencetak bagian atas dan satu untuk bagian bawah.
4. Loop pertama (Bagian Atas) mencetak huruf dari 'A' hingga karakter sesuai dengan baris saat ini. Pada setiap iterasi, program akan menghitung jumlah huruf berdasarkan rumus 2 * i - 1 (baris pertama memiliki 1 huruf, baris kedua memiliki 3 huruf, dan seterusnya sama seperti pola bilangan ganjil).
5. Loop kedua mencetak huruf untuk bagian bawah, dengan jumlah huruf yang menurun setelah mencapai puncak (jumlah huruf di baris-baris berikutnya berkurang dari 9 kembali ke 1).
6. Progam menggunakan karakter ASCII dimana kode ASCII memiliki karakter berurutan setelah huruf 'Z', sehingga bisa digunakan untuk mengikuti pola yang lebih besar dari alfabet. Jika `i <= 26`, maka karakter yang digunakan adalah huruf 'A' hingga 'Z' (berdasarkan kode ASCII mulai dari 65 untuk 'A'). Jika `i > 26`, program mengurangi 26 dari indeks untuk berlanjut ke karaker berikutnya dalam tabel ASCII, seperti `[`,`\`,`]`, dan seterusnya.
7. Setiap baris yang dicetak di-"center"-kan menggunakan metode .center(). Untuk memastikan spasi yang cukup, lebar maksimal dikalikan dengan 2, sehingga ada ruang lebih untuk meratakan pola ditengah layar.